### ISSUE TITLE
붓꽃의 품종분류

### ISSUE THREAD

**Issue by minji-o-j** [minji-o-j/Machine-Learning#3]

**Comment by minji-o-j** [comment]
## 용어
- Task (목표)
  - 어떤 품종인지 구분해 놓은 측정 데이터를 이용해, 새로 채집한 붓꽃의 품종을 예측하는 머신러닝 모델을 만들자.
- 지도학습(Supervised learning)
  - 붓꽃의 품종을 정확하게 분류한 데이터를 가지고 있음
  - 레이블 o
- 분류문제(Classification)
  - 몇 가지 선택사항(품종) 중 하나를 선택하는 문제
- 클래스(Class)
  - 출력될 수 있는 값(붓꽃의 종류) - 3가지
- 레이블(Label)
  - 특정 데이터 포인트에 대한 출력 (특정 데이터의 품종)

**Comment by minji-o-j** [comment]
```py
X_train, X_test, y_train, y_test = train_test_split(c_data, data['binaryQuality'],test_size=0.3,random_state=0)
X_test, X_val, y_test, y_val  = train_test_split(X_test, y_test,test_size=0.5,random_state=0)
```
- train:test:valid를 70:15:15로 나눈 코드
- `train_test_split` : 데이터셋을 **섞어서 나눠준다.**
  - 데이터가 모여있는 경우 섞지 않은 채로 이용하면, train set의 레이블은 모두 2가 되는 등 치우침 현상 발생 가능.
- 이 작업을 여러번 수행해도 결과가 같게 하는법?
  - `random_state`에 난수 초기값을 같은걸로 전달하면 됨
  - `random_state=42` 등.. 
- 반환되는 X_train, y_train, X_valid, 등...은 모두 **NumPy** 배열

**Comment by minji-o-j** [comment]
## 데이터 살펴보기
### 왜 하는걸까?
- 머신러닝 모델 만들기 전에 **머신러닝 없이도 풀 수 있는 문제가 아닌지, 필요 정보 누락되지 않았는지** 에 대한 데이터 조사
- **비정상적인 값**이나 **특이한 값**을 찾을 수도 있음
  - ex) 단위의 통일이 안되어있는 경우
  - 데이터 일관성X, 이상한 값

### 하는 방법
- **시각화**
- 산점도 이용
- 3개 이상의 특성을 표현하기 어렵다.

**Comment by minji-o-j** [comment]
## k-Nearest Neighbors, k-NN
- 새로운 데이터 포인트에 가장 가까운 ‘k개’의 이웃을 찾음  
<br>

![image](https://user-images.githubusercontent.com/45448731/85199156-6141fe00-b328-11ea-9f3d-cecb78101feb.png)

- `n_neighbors=1` 가장 가까운 1개의 이웃 찾음

- `.fit` : 훈련 데이터셋으로부터 모델을 만들기 위해서는 knn 객체의 `fit` method 사용 필요
- knn 객체 자체가 변경됨(데이터로 훈련됨)

**Comment by minji-o-j** [comment]
## 예측하기
---
## 모델 평가하기
